test_namespace_pkg/existing/__init__.py

```diff
@@ -0,0 +1,7 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
```

Comment:
We can't put it in `tests/` as `tests/` is a normal package.. :(

Would you mind making the `namespace` repository?

Reply:
Can we launch the tests from ``tests/data/`` instead of from the root so the test suite do not realize that it's inside a package ? Depending on another repo is not ideal I think you were right to want to have the test data here.

Reply:
No, for this test to work there shouldn't be a `__init__` in between the package and the `cwd`...

Reply:
I searched for ``os.chdir`` in the code base and we seem to have a number of test fixture of context manager to handle this, in particular ``            with _test_cwd():`` in test_self.py. I think we could even unify this at some point because there seem to be quite a bit of duplication.  Wouldn't that approach work ?

```python
@contextlib.contextmanager
def _test_cwd(new_cwd: Path | str) -> Generator[None, None, None]:
    original_dir = os.getcwd()
    os.chdir(new_cwd)
    try:
        yield
    finally:
        os.chdir(original_dir)



with _test_cwd("tests/data/namespace/"):
    ...
```

Reply:
Just quickly touching in: just got some bad family news. Not sure if I'll be able to devote time to programming over the next week.

If somebody could take a look at this final comment and get this PR merged for the beta that would be much appreciated!

Reply:
Of course, and sorry to hear that. Be well.

Reply:
Sorry to hear that Daniel, take care of yourself.

Reply:
I checked, but this doesn't work... We pass both with and without changes with the current tests. I really think we can't do much else than create a new root directory or a new package.

I believe this is due to how `pytest` adds `tests` to `sys.path` or something. Make all attempts to mix `namespace` with `normal` packages in a test directory fail.

Reply:
A new test file under `tests/regrtest_data/` fixes this. Maybe preferable?

```diff
diff --git a/tests/regrtest_data/test_namespace_package.py b/tests/regrtest_data/test_namespace_package.py
new file mode 100644
index 00000000..aa9194ad
--- /dev/null
+++ b/tests/regrtest_data/test_namespace_package.py
@@ -0,0 +1,11 @@
+from io import StringIO
+from pylint.reporters.text import TextReporter
+from pylint.testutils._run import _Run as Run
+
+
+def test_namespace_package_sys_path() -> None:
+    """Test that we do not append namespace packages to sys.path."""
+    pylint_output = StringIO()
+    reporter = TextReporter(pylint_output)
+    Run(["tests/regrtest_data/test_namespace_pkg/"], reporter=reporter, exit=False)
+    assert "Module import itself" not in pylint_output.getvalue()
diff --git a/tests/regrtest_data/test_namespace_pkg/else/__init__.py b/tests/regrtest_data/test_namespace_pkg/else/__init__.py
new file mode 100644
index 00000000..e69de29b
diff --git a/tests/regrtest_data/test_namespace_pkg/existing/__init__.py b/tests/regrtest_data/test_namespace_pkg/existing/__init__.py
new file mode 100644
index 00000000..1b93e7d8
--- /dev/null
+++ b/tests/regrtest_data/test_namespace_pkg/existing/__init__.py
@@ -0,0 +1,3 @@
+from existing import my_func  # type: ignore[attr-defined]
+
+my_func()
```

Reply:
Hm, that's weird. I thought I checked this and that this wouldn't currently fail. I'll try this `diff` tomorrow. Thanks @jacobtylerwalls for the additional investigation! 😄